### ISSUE TITLE
Let users change their information

### ISSUE THREAD

**Issue by jenrik** [f-klubben/stregsystemet#234]
Currently user can't change their name, email, gender or if whether want spam on their own and have to contact FIT if they want it changed.

I suggest we implement a settings page reachable from from the user info page. Perhaps we could also allow users to change their username if it's available.

**Comment by jones-g** [comment]
I support the option to change name, email and gender but the Want Spam "feature" should be included.

Note that our primary communications channel (unless otherwise specified) is still email. We send out our legal notifications about general assembly to our members this way. We are required to send this to all active members and we use the fembers list to track this. If people do not want to receive mails that can have their accounts deactivated instead, this will seize emails as inactive users are removed from the fembers list. Otherwise the compromise is that they will need to be on the fembers list if their account should remain active. This has been discussed many times before. The Want Spam field should not be used!

**Comment by cogitantium** [comment]
I agree that this is a feature we should implement. However, since we do not have any real access control (and shouldn't, in my opinon) we should be careful with how we allow and track changes. I suggest that any profile changes are notified to the registered user email and for email-changes, both old and new email gets notified. 
This ensures that any abuse is easily identified by the users.

**Comment by jones-g** [comment]
@cogitantium Excellent point - I'd did not consider this at all. This is of course also an issue. Could perhaps be bypassed with a password that Treo can set on the user to allow these types of changes if need be. But not require it for purchases.

**Comment by cogitantium** [comment]
I'm not sure I understand; would this require FIT to set a password individually on each user on request to enable these updates? I do find it a bit ironic that we'd require authentication for changing _virtually_ unimportant values but none for doing purchases.

I believe we should try to limit the amount of over-engineering and complexity we introduce, especially any that increases the need for FIT to be an active part of the process. Since we've historically operated Stregsystemet on a high level of trust maybe we should extend this trust to the changing of user details.

In other words; I think simply allowing any member to change details from the userinfo page (requires only knowing their username) and sending email notifications for any changes would be a viable starting-point. If we find that abuse happens, we can add fail2ban-inspired rules if the problem is widespread or a read-only flag to the user model to disallow changes if only some users are experiencing abuse.
